dbbuild/lib/source.py

```diff
@@ -164,6 +166,8 @@ def all_markers(self):
     def update_marker_names(self):
         markers_by_locus = defaultdict(Locus)
         for marker in self.all_markers():
+            if marker.name in self.interval_index.mergeables:
+                marker.name = self.interval_index.mergeables[marker.name]
```

Comment:
Renaming of merged microhaps needs to happen before updating ambiguous marker names.